docs/madr/decisions/033-meshhttproute-meshgateway.md

```diff
@@ -0,0 +1,104 @@
+# `MeshHTTPRoute` for `MeshGateway`
+
+- Status: accepted
+
+Technical Story: #8142
+
+## Context and Problem Statement
+
+We have two very similar resources, `MeshHTTPRoute` and `MeshGatewayRoute` that
+both configure route matching, filtering and redirecting semantics.
+This MADR proposes adapting `MeshHTTPRoute` for use with
+`MeshGateway.`
+
+## Decision Drivers <!-- optional -->
+
+* Fewer CRDs
+* Parity with `MeshGatewayRoute`
+
+## Considered Options
+
+- Keep two resources
+- Use `MeshHTTPRoute`
+
+## Decision Outcome
+
+Chosen option:
+
+- Adapt `MeshHTTPRoute`
+- Add `spec.to[].hostnames`
+- Allow `spec.to[].targetRef` to be empty
+- Require `backendRefs` to be set
+- Don't support policy matching on `MeshHTTPRoute` if it attaches to a gateway
+
+### Hostname
+
+With `MeshGatewayRoute` and `HTTPRoute`, users can set hostnames to only apply
+the route rules to requests going to the given hosts.
+We don't have this possibility with `MeshHTTPRoute`, so we can either:
+
+- add `hostnames` to the `MeshHTTPRoute`, likely directly under `spec.to`
+  entries
+- have users on each rule do hostname matching using the header match on `host`
+
+This MADR proposes adding `spec.to[].hostnames`. Hostname matching is in some sense the
+"highest level" matching possible with HTTP and it's likely that all rules of a
+given resource are intended to apply to a given hostname set, whether that's a
+specific hostname, a wildcard pattern or to all hostnames.
+
+The second option might be feasible but it only makes sense if the cost of having
+host matching separate is very high. It also differs from Gateway API, which is
+a disadvantage.
```

Comment:
+1 for option 1. I think even if hostnames had an effect on regular routes that wouldn't be a problem. So to be clear this would be part of the `Match` part of the spec? Maybe an example would be good here.

Reply:
Good point I'll add some more examples.

Reply:
added `hostnames` to the example. it's not in match. it's next to `rules`.